sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateOperation.cs

```diff
@@ -69,11 +70,24 @@ public override Response UpdateStatus(CancellationToken cancellationToken = defa
         {
             if (!_completed)
             {
-                Response<CertificateOperationProperties> pollResponse = _client.GetPendingCertificate(Properties.Name, cancellationToken);
+                Response<CertificateOperationProperties> pollResponse = _client.GetPendingCertificate(Properties.Name, _requestedOperation, cancellationToken);
 
                 _response = pollResponse.GetRawResponse();
 
                 Properties = pollResponse;
+
+                // Response value should only ever be null for a delete request.
+                if (Properties is null)
+                {
+                    if (_requestedOperation != CertificateOperationRequest.Delete)
+                    {
+                        throw new InvalidOperationException($"Expected response when requested operation is {_requestedOperation}");
```

Comment:
I don't think we should ever throw from UpdateStatus except for the case when HTTP request fails.

Reply:
Value should throw instead.